### PR TITLE
Update AlternativeVerticals to use React Router

### PR DESCRIPTION
**Pull request by nmanu1**
Use `Link` to update the URL and navigate to the correct universal/vertical page, instead of showing the results on the current page.

J=SLAP-1766
TEST=manual

Click on the vertical and universal links in the alternative verticals component and check that they navigate to the correct page.

### CODE REVIEW

File: src/components/AlternativeVerticals.tsx [hunk 1]
```diff
@@ -154,10 +150,9 @@ export default function AlternativeVerticals ({
     return (
       <div className={cssClasses.categoriesText}>
         <span>View results across </span>
-        <button className={cssClasses.allCategoriesLink} 
-          onClick={actions.executeUniversalQuery}>
+        <Link className={cssClasses.allCategoriesLink} to={`/?query=${query}`}>
           <span>all search categories.</span>
```
Comment (yen-tt) [inline src/components/AlternativeVerticals.tsx:154]: is the span tag needed? can we have literal string directly in the Link component instead

File: src/components/AlternativeVerticals.tsx [hunk 2]
```diff
@@ -138,14 +138,10 @@ export default function AlternativeVerticals ({
   function renderSuggestion(suggestion: VerticalSuggestion) {
     return (
       <li key={suggestion.verticalKey} className={cssClasses.suggestion}>
-        <button className={cssClasses.suggestionButton}
-          onClick={() => {
-            actions.setVertical(suggestion.verticalKey);
-            actions.executeVerticalQuery();
-          }}>
+        <Link className={cssClasses.suggestionButton} to={`/${suggestion.verticalKey}?query=${query}`}>
```
Comment (yen-tt) [inline src/components/AlternativeVerticals.tsx:141]: I don't think `?query=${query}` part is needed for the router

File: src/components/AlternativeVerticals.tsx (hunk 1 above)
Comment (nmanu1) [reply src/components/AlternativeVerticals.tsx]: you're right, it's not needed so I removed it

File: src/components/AlternativeVerticals.tsx (hunk 2 above)
Comment (nmanu1) [reply src/components/AlternativeVerticals.tsx]: removed!

**Review by yen-tt — APPROVED** [review]

**Pull request merged**